core/DependencyInjection/LegacyFileLoader.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ */
+
+declare(strict_types=1);
+
+namespace oat\generis\model\DependencyInjection;
+
+use Closure;
+use oat\oatbox\config\ConfigurationService;
+use oat\oatbox\Configurable;
+use oat\oatbox\service\ConfigurableService;
+use oat\oatbox\service\ServiceFactoryInterface;
+use oat\tao\model\OntologyClassService;
+use oat\taoDeliveryRdf\model\DeliveryAssemblyService;
+use oat\taoGroups\models\GroupsService;
+use oat\taoLti\models\classes\ConsumerService;
+use oat\taoOutcomeUi\model\ResultsService;
+use oat\taoTestTaker\models\TestTakerService;
+use ReflectionClass;
+use SplFileInfo;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Loader\FileLoader;
+use Symfony\Component\DependencyInjection\Reference;
+use taoItems_models_classes_ItemsService;
+use taoQtiTest_models_classes_QtiTestService;
+use taoTests_models_classes_TestsService;
+
+class LegacyFileLoader extends FileLoader
+{
+    private const LEGACY_CONFIGURABLE_CLASS_LIST = [
+        ConfigurableService::class,
+        Configurable::class,
+        ConfigurationService::class
+    ];
+
+    private const UNSUPPORTED_LEGACY_CLASSES = [
+        taoTests_models_classes_TestsService::class,
+        taoItems_models_classes_ItemsService::class,
+        TestTakerService::class,
+        GroupsService::class,
+        ResultsService::class,
+        ConsumerService::class,
+        DeliveryAssemblyService::class,
+        taoQtiTest_models_classes_QtiTestService::class,
```

Comment:
Even knowing that is MVP, I feel so bad merging this into development. Do you have any ideas on how can that be avoided without significant effort?

Reply:
Fixed, now the base container falls back to ServiceManager, so we can load any class from container from now :)